chalk-ir/src/lib.rs

```diff
@@ -1731,6 +1845,16 @@ pub struct EqGoal<I: Interner> {
 
 impl<I: Interner> Copy for EqGoal<I> where I::InternedGenericArg: Copy {}
 
+/// Subtype goal: tries to prove that `a` is a subtype of `b`
+#[derive(Clone, PartialEq, Eq, Hash, Fold, Visit, Zip)]
+#[allow(missing_docs)]
+pub struct SubtypeGoal<I: Interner> {
```

Comment:
The alternative here was to add a `variance: Variance` field to `EqGoal` and change it to `RelateGoal`